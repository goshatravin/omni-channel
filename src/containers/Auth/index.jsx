import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import ErrorComponent from '../../components/ErrorComponents';
import InputComponent from '../../components/InputComponent';
import ButtonComponent from '../../components/ButtonComponent';
import LogoComponent from '../../components/LogoComponents';
import { authUser } from '../../store/actions/loginAction';
import errorAction from '../../store/actions/ui/errorAction';
import Div from '../../components/layout/Flexbox';
import FormBox from '../../components/layout/Form';

const Auth = (props) => {
  const { dispatch, loggedIn, error } = props;
  const [login, setLogin] = useState('');
  const [password, setPassword] = useState('');
  const [submitted, setSubmitted] = useState(false);

  const submitForm = (e) => {
    e.preventDefault();
    setSubmitted(true);
    if (login.length > 2 && password.length > 2) {
      dispatch(authUser(login, password));
    } else {
      dispatch(errorAction());
    }
  };
  if (!loggedIn) {
    return (
      <Div className="login" direction="column">
        <LogoComponent />
        <ErrorComponent submitted={submitted} error={error} />
        <FormBox
          onSubmit={(e) => {
            submitForm(e);
          }}
        >
          <InputComponent
            error={error}
            validation={submitted}
            className="input-form"
            inputType="text"
            name="name"
            placeholder="Логин"
            controlFunc={(e) => {
              setLogin(e.target.value);
            }}
            value={login}
          />
          <InputComponent
            error={error}
            validation={submitted}
            className="input-form"
            inputType="password"
            name="password"
            placeholder="Пароль"
            controlFunc={(e) => {
              setPassword(e.target.value);
            }}
            value={password}
          />
          <ButtonComponent
            className="btn"
            type="submit"
            name="submit"
            value="Войти"
          />
        </FormBox>
      </Div>
    );
  }
  return <Redirect to="/main" />;
};

Auth.propTypes = {
  dispatch: PropTypes.func,
  loggedIn: PropTypes.bool.isRequired,
  error: PropTypes.shape({
    error: PropTypes.bool,
    status: PropTypes.string,
    text: PropTypes.string,
    value: PropTypes.string,
  }),
};

Auth.defaultProps = {
  error: null,
  dispatch: () => {},
};

const mapStateToProps = (state) => {
  console.log(state);
  const { loggedIn, error } = state.loginReducer;
  return {
    loggedIn,
    error,
  };
};

const connectedAuth = connect(mapStateToProps)(Auth);
export default connectedAuth;
