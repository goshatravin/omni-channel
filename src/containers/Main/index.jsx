import React from 'react';
import axios from 'axios';

const Dashboard = (props) => {
  console.log(props);
  const myF = () => {
    console.log('123');
    axios({
      method: 'delete',
      url: 'http://localhost:4000/api/pages/logout',
      withCredentials: true,
    }).then(props.history.push('/login'));
  };
  return (
    <div className="dashboard">
      <button onClick={myF}>qweqwewewqewq</button>
    </div>
  );
};

export default Dashboard;
