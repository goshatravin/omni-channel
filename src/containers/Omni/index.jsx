import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import io from 'socket.io-client';
import { getTicket, getTicketInfo } from '../../store/actions/omniAction';
import { getReference } from '../../store/actions/referenceAction';
import ticketUpdate from '../../store/actions/socketAction';
import TasksComponent from '../../components/TasksComponents';
import InputComponent from '../../components/InputComponent';
import SearchComponent from '../../components/SearchComponent';
import ChatComponent from '../../components/ChatComponent';
import ChatHeaderComponent from '../../components/ChatHeaderComponent';
import {
  Grid,
  Col,
  Row,
  Block,
  BlockOverflow,
  FlexBlock,
} from '../../components/layout/Grid';

export const Dashboard = (props) => {
  console.log(props);
  const {
    dispatch,
    ticket,
    ticketIsLoading,
    ticketInfo,
    reference,
    referenceIsLoading,
    ticketInfoIsLoading,
  } = props;
  const [chatMessage, setChatMessage] = useState('');
  const [headerInfo, setHeaderInfo] = useState({});
  const ENDPOINT = 'http://localhost:8084';
  const [page, setPage] = useState(1);
  const [searchValue, setSearchValue] = useState('');

  useEffect(() => {
    const socket = io(ENDPOINT);
    socket.emit('omnichannelConnect');
    socket.on('omnichannelNotify', (data) => {
      dispatch(ticketUpdate(data));
    });
  }, []);

  useEffect(() => {
    dispatch(getTicket('/ticket'));
    dispatch(getReference('/reference_book/channel_type'));
  }, [page]);

  useEffect(() => {
    console.log(searchValue);
  }, [searchValue]);

  const ticketHandler = (data) => {
    setHeaderInfo(data);
    dispatch(getTicketInfo(`/signal/${data.ticket_id}`));
  };
  if (!ticketIsLoading && !referenceIsLoading && ticket && reference) {
    return (
      <Grid>
        <Row>
          <Col size={1}>
            <Block className="Search-Block">
              <SearchComponent searchHandler={setSearchValue} />
            </Block>
            <BlockOverflow>
              <TasksComponent
                ticketValue={ticket}
                referenceValue={reference}
                ticketHandler={ticketHandler}
              />
            </BlockOverflow>
          </Col>
          <Col size={2}>
            {ticketInfo && !ticketInfoIsLoading ? (
              <FlexBlock>
                <ChatHeaderComponent data={headerInfo} />
                <ChatComponent data={ticketInfo} />
                <InputComponent
                  className="input_chat"
                  inputType="input"
                  placeholder="Сообщение"
                  name="input"
                  width="450px"
                  controlFunc={(e) => {
                    setChatMessage(e.target.value);
                  }}
                  value={chatMessage}
                />
              </FlexBlock>
            ) : (
              ''
            )}
          </Col>
          <Col size={4}></Col>
        </Row>
      </Grid>
    );
  }
  return <p>loading</p>;
};

const mapStateToProps = (state) => {
  /**
   * mapStateToProps() returns a redux state
   * @ticket - contains information about new appeals
   * @ticketIsLoading - constant which shows loading of ticket
   * @ticketInfo - contains ticket information such as chating history
   * @ticketError - contains information about ticket errors
   * @ticketInfoIsLoading - constant which shows loading of ticket chat history
   * @reference  - constains information about channel_type (viber, telegram...)
   * @referenceIsLoading - constant which shows loading of reference
   */
  const {
    ticket,
    ticketIsLoading,
    ticketInfo,
    ticketError,
    ticketInfoIsLoading,
  } = state.omniReducer;
  const { reference, referenceIsLoading } = state.referenceReducer;
  return {
    ticket,
    ticketIsLoading,
    ticketInfo,
    reference,
    referenceIsLoading,
    ticketError,
    ticketInfoIsLoading,
  };
};

const connectedDashboard = connect(mapStateToProps)(Dashboard);
export default connectedDashboard;
