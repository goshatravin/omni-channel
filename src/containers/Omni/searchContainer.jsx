import React, { useState, useEffect } from 'react';
import SearchComponent from '../../components/SearchComponent';

const SearchContainer = (props) => {
  console.log(props);
  const searchHandler = (e) => {
    console.log(e);
    props.setSearchValue(e);
  };
  return <SearchComponent searchHandler={searchHandler} />;
};

export default SearchContainer;
