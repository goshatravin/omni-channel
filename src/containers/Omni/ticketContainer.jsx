import React, { useState, useEffect, useRef, useCallback } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import TicketComponent from '../../components/TicketComponents'; // Dump component for rendering tickets
// import SearchComponent from '../../components/SearchComponent'; // Dump component from rendering search bar
import { getTicket } from '../../store/actions/omniAction';

// const TaskDiv = styled.div`
//   cursor: pointer;
//   padding: 1rem 0.5rem;
//   font-size: 14px;
//   width: 315px;
//   height: 100px;
//   background: yellow;
//   margin-bottom: 1rem;
//   margin-right: 0.5rem;
//   display: flex;
//   flex-direction: column;
//   justify-content: space-between;
//   color: ${(props) => props.theme.colors.text};
//   border: 1px solid ${(props) => props.theme.colors.input_border};
//   border-radius: 10px;
//   background: white;
//   outline: none;
//   &::placeholder {
//     color: ${(props) => props.theme.colors.input_placeholder};
//   }
// `;
const TicketContainer = (props) => {
  console.log(props);
  const { dispatch, ticketIsLoading, ticket, ticketError, hasMore } = props;
  console.log(ticket);
  const [page, setPage] = useState(1); // setup starting page for calling ticket api
  const [searchValue, setSearchValue] = useState('');

  const observer = useRef();
  const lastTicektRef = useCallback(
    (node) => {
      if (ticketIsLoading) return;
      if (observer.current) observer.current.disconnect();
      observer.current = new IntersectionObserver(
        (entries) => {
          if (entries[0].isIntersecting && hasMore) {
            setPage((prevPageNumber) => prevPageNumber + 1);
          }
        },
        { threshold: 1 }
      );
      if (node) observer.current.observe(node);
    },
    [ticketIsLoading, hasMore]
  );
  //
  useEffect(() => {
    dispatch(getTicket('/ticket', searchValue, page));
  }, [page]);

  return (
    <>
      {/* <SearchComponent searchHandler={setSearchValue} /> */}
      <TicketComponent
        lastHendler={lastTicektRef}
        ticket={ticket}
        loading={ticketIsLoading}
        error={ticketError}
      />
      {/* {ticket.map((item, index) => {
        if (ticket.length === index + 1) {
          return (
            <TaskDiv ref={lastTicektRef} key={item.ticket_id}>
              {item.created_by}
            </TaskDiv>
          );
        } else {
          return <TaskDiv key={item.ticket_id}>{item.created_by}</TaskDiv>;
        }
      })}
      <div>{ticketIsLoading && 'Loading...'}</div>
      <div>{ticketError && 'Error'}</div> */}
    </>
  );
  // return !ticketIsLoading && !ticketError
  //   ? ticket.map((item, index) => {
  //       if (ticket.length === index + 1) {
  //         return (
  //           <TaskDiv ref={lastTicektRef} key={item.ticket_id}>
  //             {item.created_by}
  //           </TaskDiv>
  //         );
  //       }
  //       return <TaskDiv key={item.ticket_id}>{item.created_by}</TaskDiv>;
  //     })
  //   : 'Loading...';
};

const mapStateToProps = (state) => {
  const {
    ticket,
    ticketError,
    ticketInfo,
    ticketInfoIsLoading,
    ticketIsLoading,
    hasMore,
  } = state.omniReducer;
  return {
    ticket,
    ticketError,
    ticketInfo,
    ticketInfoIsLoading,
    ticketIsLoading,
    hasMore,
  };
};

const connectedDashboard = connect(mapStateToProps)(TicketContainer);
export default connectedDashboard;
