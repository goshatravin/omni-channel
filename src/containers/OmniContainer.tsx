import React, { useState } from 'react';
import { connect } from 'react-redux';
import {
  Wrapper,
  SectionTicket,
  SectionMessage,
  SectionInfo
} from '../components/Grid';
import OmniTicketComponent from '../components/OmniTicketComponent';
import OmniSearchComponent from '../components/OmniSearchComponent';
import tickets from '../__test__/tickets.json';

type OmniContainerProps = {};
const OmniContainer: React.FC<OmniContainerProps> = (props) => {
  console.log(props);
  const [searchValue, setSearchValue] = useState<string>('');
  const [currentBtn, setCurrentBtn] = useState<string>('');
  const [currentTicket, setCurrentTicket] = useState<string>('');
  const switchBtn = (event: string, filter: string): void => {
    setSearchValue(filter);
    setCurrentBtn(event);
  };
  const setValue = (event: string): void => {
    setSearchValue(event);
    setCurrentBtn('');
  };
  const ticketInfoRequest = (e: {
    assigned_to: string | null;
    case: string | null;
    case_status: string | null;
    channel: string | null;
    created_at: string | null;
    created_by: string | null;
    customer: string | null;
    insured_dob: string | null;
    latest_signal: string | null;
    remark: string | null;
    status_type: string | null;
    ticket_id: string;
    updated_at: string | null;
  }) => {
    const id = e.ticket_id;
    setCurrentTicket(id);
  };
  return (
    <Wrapper>
      <SectionTicket>
        <OmniSearchComponent
          searchValue={searchValue}
          setValue={setValue}
          switchBtn={switchBtn}
          currentBtn={currentBtn}
        />
        <OmniTicketComponent
          tickets={tickets}
          currentTicket={currentTicket}
          ticketInfoRequest={ticketInfoRequest}
        />
      </SectionTicket>
      <SectionMessage />
      <SectionInfo />
    </Wrapper>
  );
};
const mapStateToProps = (state: any) => state;

const connectedOmniContainer = connect(mapStateToProps)(OmniContainer);
export default connectedOmniContainer;
