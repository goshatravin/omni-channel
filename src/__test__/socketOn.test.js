import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow, mount } from 'enzyme';
import React from 'react';
import { Dashboard } from '../containers/Omni';

// const useEffect = jest.spyOn(React, 'useEffect').mockImplementation(() => {});
describe('test omni container', () => {
  const storeq = {
    ticket: {},
    ticketIsLoading: false,
    referenceIsLoading: false,
    reference: [],
  };

  it('should render main pages with props', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Dashboard {...storeq} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('should render main pages without props', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Dashboard />);
    const result = renderer.getRenderOutput();
    expect(result.props.children).toEqual('loading');
    expect(result).toMatchSnapshot();
  });
});

// !ticketIsLoading && !referenceIsLoading && ticket && reference;
