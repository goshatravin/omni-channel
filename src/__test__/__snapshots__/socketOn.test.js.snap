// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test omni container should render main pages with props 1`] = `
<ForwardRef(styled.div)>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      size={1}
    >
      <ForwardRef(styled.div)
        className="Search-Block"
      >
        <SearchComponent
          placeholder="Введите для поиска"
          searchHandler={[Function]}
        />
      </ForwardRef(styled.div)>
      <ForwardRef(styled.div)>
        <TasksComponents
          referenceValue={Array []}
          results={Array []}
          ticketHandler={[Function]}
          ticketValue={Object {}}
        />
      </ForwardRef(styled.div)>
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      size={2}
    >
      
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)
      size={4}
    />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;

exports[`test omni container should render main pages without props 1`] = `
<p>
  loading
</p>
`;
